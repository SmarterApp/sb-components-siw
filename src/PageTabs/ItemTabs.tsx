import * as React from "react";

/**
 * Names of the different tab types.
 * @type {Tabs}
 */
export type Tabs = "viewer" | "rubric" | "information";

/**
 * ItemTabsProps props
 * @export
 * @interface ItemTabsProps
 */
export interface ItemTabsProps {
  changedTab: (tab: Tabs) => void;
  selectedTab: Tabs;
  showRubricTab?: boolean;
}

/**
 * ItemTabs displays options for view state of a selected item.
 * @export
 * @class ItemTabs
 * @extends {React.Component<ItemTabsProps, {}>}
 */
export class ItemTabs extends React.Component<ItemTabsProps, {}> {
  private select: Tabs;

  constructor(props: ItemTabsProps) {
    super(props);

    this.state = {
      selectedTab: this.props.selectedTab
    };
  }

  /**
   * event handler for view state.
   */
  onClick = (selectedVal: Tabs) => {
    this.props.changedTab(selectedVal);
  };

  /**
   * renders individual tab components
   * @param {string} label
   * @param {Tabs} tabType
   * @returns
   */
  renderTab(label: string, tabType: Tabs) {
    return (
      <li
        role="button"
        className={
          this.props.selectedTab === tabType ? "nav-item active" : "nav-item"
        }
        onClick={() => this.onClick(tabType)}
      >
        <a className="nav-link">{label}</a>
      </li>
    );
  }

  render() {
    let rubricTab;
    if (this.props.showRubricTab && this.props.showRubricTab === true) {
      rubricTab = this.renderTab("Rubric and Exemplar", "rubric");
    }

    return (
      <ul className="nav nav-tabs">
        {this.renderTab("Item Viewer", "viewer")}
        {rubricTab}
        {this.renderTab("Item Information", "information")}
      </ul>
    );
  }
}
