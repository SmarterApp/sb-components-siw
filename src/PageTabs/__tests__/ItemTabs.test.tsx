import * as React from "react";
import * as ReactDOM from "react-dom";
import { ItemTabs, Tabs } from "../ItemTabs";
import { shallow, mount, render } from "enzyme";

describe("PageTabs", () => {
  const tabs: Tabs[] = ["viewer", "rubric", "information"];

  const props = {
    changedTab: jest.fn((tab: Tabs) => {
      return undefined;
    }),
    selectedTab: tabs[0]
  };

  it("matches snapshot for each tab with no rubric", () => {
    const wrapper = shallow(<ItemTabs {...props} />);
    tabs.forEach(tab => {
      wrapper.setProps({ selectedTab: tab });
      expect(wrapper).toMatchSnapshot();
    });
  });

  it("matches snapshot for each tab with rubric", () => {
    const wrapper = shallow(<ItemTabs {...props} showRubricTab={true} />);
    tabs.forEach(tab => {
      wrapper.setProps({ selectedTab: tab });
      expect(wrapper).toMatchSnapshot();
    });
  });

  it("changes tabs", () => {
    const wrapper = shallow(<ItemTabs {...props} />);
    wrapper
      .find("ul.nav-tabs")
      .childAt(0)
      .simulate("click");
    expect(props.changedTab).toHaveBeenCalled();
  });
});
