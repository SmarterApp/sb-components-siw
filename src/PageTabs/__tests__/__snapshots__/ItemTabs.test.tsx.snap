// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTabs matches snapshot for each tab with no rubric 1`] = `
<ul
  className="nav nav-tabs"
>
  <li
    className="nav-item active"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Item Viewer
    </a>
  </li>
  <li
    className="nav-item"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Item Information
    </a>
  </li>
</ul>
`;

exports[`PageTabs matches snapshot for each tab with no rubric 2`] = `
<ul
  className="nav nav-tabs"
>
  <li
    className="nav-item"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Item Viewer
    </a>
  </li>
  <li
    className="nav-item"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Item Information
    </a>
  </li>
</ul>
`;

exports[`PageTabs matches snapshot for each tab with no rubric 3`] = `
<ul
  className="nav nav-tabs"
>
  <li
    className="nav-item"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Item Viewer
    </a>
  </li>
  <li
    className="nav-item active"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Item Information
    </a>
  </li>
</ul>
`;

exports[`PageTabs matches snapshot for each tab with rubric 1`] = `
<ul
  className="nav nav-tabs"
>
  <li
    className="nav-item active"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Item Viewer
    </a>
  </li>
  <li
    className="nav-item"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Rubric and Exemplar
    </a>
  </li>
  <li
    className="nav-item"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Item Information
    </a>
  </li>
</ul>
`;

exports[`PageTabs matches snapshot for each tab with rubric 2`] = `
<ul
  className="nav nav-tabs"
>
  <li
    className="nav-item"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Item Viewer
    </a>
  </li>
  <li
    className="nav-item active"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Rubric and Exemplar
    </a>
  </li>
  <li
    className="nav-item"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Item Information
    </a>
  </li>
</ul>
`;

exports[`PageTabs matches snapshot for each tab with rubric 3`] = `
<ul
  className="nav nav-tabs"
>
  <li
    className="nav-item"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Item Viewer
    </a>
  </li>
  <li
    className="nav-item"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Rubric and Exemplar
    </a>
  </li>
  <li
    className="nav-item active"
    onClick={[Function]}
    role="button"
  >
    <a
      className="nav-link"
    >
      Item Information
    </a>
  </li>
</ul>
`;
